### ISSUE TITLE
fake_quant_with_min_max_vars  can only get five significant figures instead of six

### ISSUE THREAD

**Issue by xueyedamo521** [tensorflow/tensorflow#48223]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): NO
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: NO
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.4.1
- Python version: 3.7.3
- Bazel version (if compiling from source): NO
- GCC/Compiler version (if compiling from source): NO
- CUDA/cuDNN version: 10.0
- GPU model and memory: Tesla V100-SXM2-32GB

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`
v2.4.0-49-22-g85c8b2a817f 2.4.1

**Describe the current behavior**
"tf.quantization.fake_quabt_with_min_max_vars" can only get five significant figures.

**Describe the expected behavior**
"tf.quantization.fake_quabt_with_min_max_vars" should get six significant figures when float32 has six significant figures.
With regard to pytorch "torch.fake_quantize_per_tensor_affine" always has six significant figures.

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

import tensorflow as tf

min_var = -10.5268
max_var = 15.1868

output_1 = tf.quantization.fake_quant_with_min_max_vars(0.27, min_var, max_var, num_bits=8, narrow_range=True)
print(output_1 ) #output_1 is 0.3037033

output_2 = tf.quantization.fake_quant_with_min_max_vars(-0.27, min_var, max_var, num_bits=8, narrow_range=True)
print(output_2 ) #output_2 is -0.30370426

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by xueyedamo521** [comment]
"fake_quant_with_min_max_vars " is  implemented using XLA. For accuracy I wonder why it is different from pytorch which can ensure six significant figures.

**Comment by Saduf2019** [comment]
@xueyedamo521 
We see that you are using an old version of tensorflow, there is no support for tf 1.x hence please upgrade to 2.x and let us know.

**Comment by xueyedamo521** [comment]
output_1 = tf.quantization.fake_quant_with_min_max_vars(0.27, min_var, max_var, num_bits=8, narrow_range=True)
print(output_1 ) #output_1 is 0.3037033

the result is error. 
scale = (max_var-min_var)/(255-1) #should be 0.1012346, and float32 can guarantee six significant figures, "6" can not be ensured.
the fake quant result of "0.27" should be "scale*3". Observely, TF is not right with only five significant figures.

**Comment by xueyedamo521** [comment]
@Saduf2019

**Comment by Saduf2019** [comment]
@ymodak 
I am able to replicate the issue reported please find the [gist here](https://colab.research.google.com/gist/Saduf2019/48b063c91a696b1d84da97d36d61eb3e/untitled583.ipynb).
Thanks!

**Comment by xueyedamo521** [comment]
@ymodak 
Looking forward to any reply from you，thanks.

**Comment by sanatmpa1** [comment]
I tried reproducing the issue in `2.6.0` and in nightly i.e `2.8.0-dev20211102`, the results are still similar to the one reported in the issue. Please find the [gist here](https://colab.sandbox.google.com/gist/sanatmpa1/3231a6791ba0dee202ba0e7645a580a8/untitled583.ipynb). Thanks!

**Comment by sushreebarsa** [comment]
@xueyedamo521 
I tried to replicate the issue in colab using TF v2.8.0 and faced a different outcome .Could you please have a look at the gist [here](https://colab.research.google.com/gist/sushreebarsa/aeac8d0dd0ab093b01eecf78f20fba7c/untitled60.ipynb) and let us know if it helps?Thanks!
